src/wix/WixToolset.Core.Burn/Bundles/BurnCommon.cs

```diff
@@ -224,8 +231,21 @@ protected bool Initialize(BinaryReader reader)
                 this.EngineSize = this.StubSize + this.UXSize;
             }
 
-            this.AttachedContainerAddress = this.ContainerCount > 1 ? this.EngineSize : 0;
-            this.AttachedContainerSize = this.ContainerCount > 1 ? BurnCommon.ReadUInt32(bytes, BURN_SECTION_OFFSET_ATTACHEDCONTAINERSIZE) : 0;
+            this.AttachedContainers.Clear();
+            uint nextAddress = this.EngineSize;
+            if (this.ContainerCount > 1)
+            {
+                for (uint i = 0; i < (this.ContainerCount - 1 /* Excluding UX */); ++i)
+                {
+                    uint sizeOffset = BURN_SECTION_OFFSET_ATTACHEDCONTAINERSIZE0 + (i * 4);
+                    uint size = BurnCommon.ReadUInt32(bytes, sizeOffset);
+                    if (size > 0)
```

Comment:
Probably not a functional change, but I find these special cases for 0 byte containers confusing. If you're trying to optimize something here, then it should be in the binder not in this low level code.